### ISSUE TITLE
MKL Error (0.3.8)

### ISSUE THREAD

**Issue by danlurie** [FCP-INDI/C-PAC#426]
I get the following error when running a pipeline. I also get a ton of crash files from the residuals workflow. By the time this error hits, it looks like most of the files have been written out (up to at least mean_functional_in_mni); so it's not clear to me at which point things break, or which steps have been completed without error. Will email my config and one of the crash files to Steve and Danny.

(I got a similar crash, possibly exactly the same one, with 0.3.7, before I tried updating in hopes of a fix).

I've previously been able to run CPAC without issues on the same machine.
```
pipeline_clg_bet_minimal_ :  pipeline_clg_bet_minimal__bbreg
compcor_ncomponents  5
Detrending and centering data
Calculating SVD decomposition of Y*Y'

MKL ERROR: Parameter 4 was incorrect on entry to DLASCL

MKL ERROR: Parameter 4 was incorrect on entry to DLASCL
Process Process-2:
Traceback (most recent call last):
  File "/home/despo/dlurie/Canopy/appdata/canopy-1.4.1.1975.rh5-x86_64/lib/python2.7/multiprocessing/process.py", line 258, in _bootstrap
    self.run()
  File "/home/despo/dlurie/Canopy/appdata/canopy-1.4.1.1975.rh5-x86_64/lib/python2.7/multiprocessing/process.py", line 114, in run
    self._target(*self._args, **self._kwargs)
  File "/home/despo/dlurie/Enthought/Canopy_64bit/User/lib/python2.7/site-packages/CPAC/pipeline/cpac_pipeline.py", line 5334, in prep_workflow
    workflow.run(plugin='MultiProc', plugin_args={'n_procs': c.numCoresPerSubject})
  File "/home/despo/dlurie/Enthought/Canopy_64bit/User/lib/python2.7/site-packages/nipype/pipeline/engine.py", line 700, in run
    runner.run(execgraph, updatehash=updatehash, config=self.config)
  File "/home/despo/dlurie/Enthought/Canopy_64bit/User/lib/python2.7/site-packages/nipype/pipeline/plugins/base.py", line 268, in run
    report_nodes_not_run(notrun)
  File "/home/despo/dlurie/Enthought/Canopy_64bit/User/lib/python2.7/site-packages/nipype/pipeline/plugins/base.py", line 92, in report_nodes_not_run
    raise RuntimeError(('Workflow did not execute cleanly. '
RuntimeError: Workflow did not execute cleanly. Check log for details
```
***Update (01/08):*** This issue seems to be specific to CompCor. Running identical pipelines with and without CompCor, the CompCor pipelines consistently crash with this error, while pipelines with WM+CSF run fine.

**Comment by jpellman** [comment]
Hi Dan,

This looks to be a very low-level issue with your configuration, and has more to do with how your computer is interacting with the processor than anything that occurs in C-PAC/Nipype code.  Could you try checking the <a href=https://software.intel.com/en-us/articles/new-in-intel-mkl-10-2>Intel Math Kernel Library website</a> to see if there are any updates that you could apply that might rectify the issue?

Best,
John Pellman

**Comment by ccraddock** [comment]
Actually this is a problem that indicates that _some_ of the wm or CSF time series that are being fed into Compcor have zero variance. I suspect that this is because the am and CSF masks are calculated from the anatomical which has different coverage and distortion a than the func. There is a fix in the next release that addresses this issue by excluding such time series from Compcor. This fix can be seen here:

https://github.com/dclark87/C-PAC/commit/240942547ee42326486cf2227177cf1e032d9e5b

**Comment by dclark87** [comment]
This error does look to be fixed and is set to be released in 0.3.9
